buku.py

```diff
@@ -2565,17 +2568,29 @@ def get_firefox_profile_name(path):
     profile : str
         Firefox profile name.
     """
+    from configparser import ConfigParser, NoOptionError
+
+    profile_path = os.path.join(path, 'profiles.ini')
+    if os.path.exists(profile_path):
+        config = ConfigParser()
+        config.read(profile_path)
+        profiles_names = [section for section in config.sections() if section.startswith('Profile')]
+        if not profiles_names:
+            return None
+        for name in profiles_names:
+            try:
+                # If profile is default
+                if config.getboolean(name, 'default'):
+                    profile_path = config.get(name, 'path')
+                    return profile_path
+            except NoOptionError:
+                continue
 
-    try:
-        for name in os.listdir(path):
-            # can be in the format nnnnnn.default or nnnnnn.default-nnnnnn...
-            if '.default' in name and os.path.isdir(os.path.join(path, name)):
-                logdbg(name)
-                return name
-    except FileNotFoundError:
-        pass
-
-    return None
+            # There is no default profile
+            return None
+    else:
+        logerr("get_firefox_profile_name(): `path' does not exist")
```

Comment:
1. Please change to `logdbg()`, we don;t want to show this to users.
2. Print `path` in the debug log, just `path` does not convey anything.

Reply:
Done.